msal/src/main/java/com/microsoft/identity/client/PublicClientApplication.java

```diff
@@ -554,51 +555,128 @@ public void run() {
+    private void getBrokerAndLocalAccount(@NonNull final String oid,
+                                          @NonNull final AccountsLoadedCallback callback) {
+        final String methodName = ":getBrokerAndLocalAccount";
 
-        if (null != realm) {
-            accountToReturn = AccountAdapter.getAccountInternal(
-                    mPublicClientConfiguration.getClientId(),
-                    mPublicClientConfiguration.getOAuth2TokenCache(),
-                    homeAccountIdentifier,
-                    realm
-            );
+        final Handler handler;
+        if (null != Looper.myLooper() && Looper.getMainLooper() != Looper.myLooper()) {
+            handler = new Handler(Looper.myLooper());
         } else {
-            com.microsoft.identity.common.internal.logging.Logger.warn(
+            handler = new Handler(Looper.getMainLooper());
+        }
+
+        //Make a background call to get account from broker
+        new BrokerMsalController().getBrokerAccounts(
+                mPublicClientConfiguration,
+                new BrokerAccountsLoadedCallback() {
+                    @Override
+                    public void onAccountsLoaded(final List<AccountRecord> accountRecords) {
+                        final List<AccountRecord> cachedAccounts = new ArrayList<>();
+                        cachedAccounts.addAll(getLocalAccounts());
+                        cachedAccounts.addAll(accountRecords);
+
+                        if (cachedAccounts.isEmpty()) {
+                            com.microsoft.identity.common.internal.logging.Logger.info(
+                                    TAG + methodName,
+                                    "No account stored in the broker cache."
+                            );
+                            handler.post(new Runnable() {
+                                @Override
+                                public void run() {
+                                    //Return null if no account found.
+                                    callback.onAccountsLoaded(null);
+                                }
+                            });
+                        }
+
+                        for (final AccountRecord accountRecord : cachedAccounts) {
+                            if (!StringUtil.isEmpty(accountRecord.getLocalAccountId())
+                                    && accountRecord.getLocalAccountId().equalsIgnoreCase(oid)) {
+                                com.microsoft.identity.common.internal.logging.Logger.info(
+                                        TAG + methodName,
+                                        "Found the matched account from broker cache."
+                                );
+
+                                handler.post(new Runnable() {
+                                    @Override
+                                    public void run() {
+                                        callback.onAccountsLoaded(AccountAdapter.adapt(accountRecord));
```

Comment:
This callback is being invoked inside a loop, surely you mean to call this only once, right?